uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js

```diff
@@ -629,6 +629,16 @@
         });
 
         describe('save leave request', function () {
+          describe('allows multiple save', function() {
+            beforeEach(function() {
+              $ctrl.submit();
+            });
+
+            it('does not allow user to submit again', function () {
+              expect($ctrl.didSubmit).toBeTruthy();
```

Comment:
actually i just realized that the reason that you are not testing for that ("when the flag is `true`, it doesn't save the request again") is because you are not doing it at all. You are simply relying on the view to hide the buttons so that the user can't click on them.

Please add also a logic to the controller that actually prevents multiple saves when the flag is `true`